### PR TITLE
busybox: update to 1.30.1.

### PR DESCRIPTION
**Pull request by CameronNemo**
busybox: enable mdev, sysklogd, and httpd applets

### CODE REVIEW

**Comment by CameronNemo** [comment]
Relates to #4289

**Comment by CameronNemo** [comment]
The base dotconfig is used for the busybox-core package. Patches are applied to bring features in to the busybox-static and busybox packages. I think this paradigm is valuable because the patches are strictly additive (functionality wise). This offers consistency and manageability.

**Comment by maxice8** [comment]
ping

**Comment by CameronNemo** [comment]
I would like to use config fragments rather than patch files. Doing this would involve fetching [merge_config.sh](https://github.com/torvalds/linux/blob/master/scripts/kconfig/merge_config.sh) then generating the dotconfig for each busybox flavor based on the fragments and busybox defaults.

**Comment by CameronNemo** [comment]
@Vaelatern I have incorporated your new busybox-huge package, with the mdev, syslogd, and klogd applets added along with httpd. The busybox and busybox-static packages maintain the same applets.

File: srcpkgs/busybox/template [hunk 1]
```diff
@@ -39,9 +53,16 @@ do_configure() {
 				"s,\(CONFIG_CROSS_COMPILER_PREFIX\).*,\1=\"${XBPS_CROSS_TRIPLET}-\",g" \
 				${t}/.config
 		fi
-		make -C $t KBUILD_SRC=${wrksrc}/src -f ${wrksrc}/src/Makefile prepare ${makejobs}
 	done
-	sed -i -e '/CONFIG_STATIC/ s/.*/CONFIG_STATIC=y/' busybox-static/.config
+
+	_patch_config busybox udhcpc ntpd
+	_patch_config busybox-static udhcpc ntpd static
+	_patch_config busybox-huge udhcpc ntpd httpd mdev sysklogd
```
Comment (CameronNemo) [inline srcpkgs/busybox/template]: This is where you can see which applet sets are used for each variant.

File: srcpkgs/busybox/template [hunk 2]
```diff
@@ -50,37 +71,41 @@ do_configure() {
...
 	vbin busybox/busybox_unstripped busybox
 	vman busybox/docs/busybox.1
 	vsv busybox-ntpd
+	vsv busybox-syslogd
+	vsv busybox-klogd
+}
+
+busybox-core_package() {
+	short_desc+=" (essential applets)"
+	conflicts="busybox>=0"
```
Comment (maxice8) [inline srcpkgs/busybox/template]: conflicts ?

File: srcpkgs/busybox/template [hunk 3]
```diff
@@ -50,37 +71,41 @@ do_configure() {
...
+		)
+	done
 }
 
 do_install() {
 	vbin busybox/busybox_unstripped busybox
 	vman busybox/docs/busybox.1
 	vsv busybox-ntpd
+	vsv busybox-syslogd
+	vsv busybox-klogd
```
Comment (maxice8) [inline srcpkgs/busybox/template]: installing them in busybox and busybox-huge ?

Comment (CameronNemo) [reply srcpkgs/busybox/template]: ah fixed, only in busybox-huge

File: srcpkgs/busybox/template (hunk 2 above)
Comment (CameronNemo) [reply srcpkgs/busybox/template]: Yeah all the paths conflict and it doesn't make sense to use more than one of them. It is a pick one type of deal.

Comment (maxice8) [reply srcpkgs/busybox/template]: ah yes i derped out, i forgot busybox is a single binary

File: srcpkgs/busybox/template [hunk 4]
```diff
@@ -83,9 +96,30 @@ do_install() {
...
 busybox-static_package() {
 	short_desc+=" (statically linked)"
 	pkg_install() {
 		vbin busybox-static/busybox busybox.static
 	}
 }
+
+busybox-huge_package() {
+	short_desc+=" (extra applets)"
+	conflicts="busybox>=0"
```
Comment (maxice8) [inline srcpkgs/busybox/template]: should conflict with busybox-core as well, no ?

File: srcpkgs/busybox/template [hunk 5]
```diff
@@ -83,9 +96,30 @@ do_install() {
 	vsv busybox-ntpd
 }
 
+busybox-core_package() {
+	short_desc+=" (essential applets)"
+	conflicts="busybox>=0"
```
Comment (maxice8) [inline srcpkgs/busybox/template]: should conflict with busybox-big as well, no ?

**Pull request merged**